webgisplugin.py

```diff
@@ -285,80 +284,76 @@ def layers_list(self):
         Returns:
             List[qgis.core.QgsMapLayer]: project's layers
         """
-        return self.iface.legendInterface().layers()
-
-    def _get_project_layers_tree(self):
-        """Returns root layer node of all project layers.
-
-        Returns:
-            webgisplugin.Node: project layers tree (root node)
-        """
-        legend_iface = self.iface.legendInterface()
-        layers_reletionship = legend_iface.groupLayerRelationship()
-        layers_root = Node('')
-        for parent_name, child_names in layers_reletionship:
-            parent = layers_root.find(parent_name)
-            if not parent:
-                parent = Node(parent_name)
-                layers_root.append(parent)
-            parent.append(*child_names)
-        return layers_root
+        return [
+            tree_layer.layer() for tree_layer in \
+            QgsProject.instance().layerTreeRoot().findLayers()]
 
     def get_project_base_layers(self):
         """Returns root layer node of all base layers.
 
         Returns:
             webgisplugin.Node: project base layers tree (root node)
         """
-        # build complete layers tree
-        layers_root = self._get_project_layers_tree()
 
-        # filter to base layers only
-        base_layers = {
-            layer.id(): layer
-            for layer in self.layers_list()
-                if self.is_base_layer_for_publish(layer)
-        }
+        def overlays_tree(tree_node):
```

Comment:
@davidtethal Please review this part.

Reply:
It's quite possible that I wrote this piece of code, and amount of changes are due to changes in Qgis api. There was iface.legendInterface().groupLayerRelationship() in Qgis 2, which was removed and instead iface.layerTreeView().layerTreeModel() was added.

Reply:
The problem was code duplication. Two almost same overlays_tree() methods. Solved by 81e08be